### PR TITLE
Git version handling

### PR DESCRIPTION
**Pull request by vkedwardli**
- Trim prefix for `current_version_str` before do the semver regex check 
- Always grab git tag with `gdxsv-` prefix
- Use semver formatting for dev build (e.g. `gdxsv-0.9.7-dev.2+dfc350a4`)
- Remove the redundant `v` prefix in the update prompt

### CODE REVIEW

File: core/gdxsv/gdxsv.cpp [hunk 1]
```diff
@@ -844,7 +844,7 @@ void Gdxsv::handleReleaseJSON(const std::string &json) {
 
     std::smatch match;
 
-    auto current_version_str = std::string(GIT_VERSION);
+    auto current_version_str = trim_prefix(std::string(GIT_VERSION));
```
Comment (inada-s) [inline core/gdxsv/gdxsv.cpp:847]: thanks

File: CMakeLists.txt [hunk 2]
```diff
@@ -37,7 +37,8 @@ endif()
 find_package(Git)
 if(GIT_FOUND AND EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/.git")
     execute_process(
-        COMMAND ${GIT_EXECUTABLE} describe --tags --always
+        COMMAND ${GIT_EXECUTABLE} describe --tags --always --match "gdxsv-*"
+        COMMAND sed "s/-\\([[:digit:]]\\{1,\\}\\)-g\\(.\\{8\\}$\\)/-dev.\\1+\\2/"
```
Comment (inada-s) [inline CMakeLists.txt]: Probably it doesn't work on windows.
You have to use general way to replace string. Please find those cmake command.

Comment (vkedwardli) [reply CMakeLists.txt:41]: Just checked, It is working in Windows.
I also tried to use `string(REGEX REPLACE ...)` command, but unfortunately it doesn't support regex quantifier `{n}`
https://stackoverflow.com/a/35845992/592511

Comment (vkedwardli) [reply CMakeLists.txt]: Thanks
Updated cmake to use `string(REGEX REPLACE ...)` command

**Comment by vkedwardli** [comment]
From the artifacts:

image
image

**Comment by inada-s** [comment]
looks good, thanks.

**Pull request merged**